kinder/pkg/build/bits/fileBits.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bits
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+
+	"k8s.io/kubeadm/kinder/pkg/extract"
+)
+
+// fileBits defines a bit installer that allows to copy a files from host machine into the node image
+type fileBits struct {
+	files []string
+}
+
+var _ Installer = &fileBits{}
+
+// NewFileBits returns a new file Installer
+func NewFileBits(files []string) Installer {
+	return &fileBits{
+		files: files,
+	}
+}
+
+// Prepare implements bits.Prepare
+func (b *fileBits) Prepare(c *BuildContext) (map[string]string, error) {
+
+	// ensure the dest path exists on host at HostBitsPath
+	dst := filepath.Join(c.HostBitsPath(), "files")
+	if err := os.Mkdir(dst, 0777); err != nil {
+		return nil, errors.Wrap(err, "failed to make bits dir")
+	}
+
+	// for each of the given file sources
+	for _, file := range b.files {
+
+		fileSrcDst := strings.Split(file, ":")
+
+		if len(fileSrcDst) != 2 {
+			return nil, errors.New("invalid value for --with-file option")
+		}
+
+		// Creates an extractor instance, that will read the file bits from src,
+		// and save it to the dest path (inside HostBitsPath)
+		e := extract.NewExtractor(
+			fileSrcDst[0], dst,
```

Comment:
This will fail if two files have the same name.
My suggestion is to use a tmp name and to maintain in a map[string]string with the mapping from source (or target) names to tmp names

Reply:
yeah, good catch. Will update according to your suggestion.

Reply:
Hey @fabriziopandini. Will I have to update the fileNameMutator in order to be able to give the files a temporary name?

Reply:
yes, you are right.
The mutator requires a new `nameOverride` field and the `Mutate` func should act accordingly.
It is also required a new option for setting the field while calling `NewExtractor` (something similar to `WithNamePrefix`)

Reply:
Thanks. I was able to get a new `nameOverride` field as you described above. For the original problem of having 2 files with the same name, we can just append the array index to come up with a temporary name.